seismicpro/src/metrics/interactive_map.py

```diff
@@ -167,17 +245,18 @@ def click(self, coords):
 class ScatterMapPlot(MetricMapPlot):
     """Construct an interactive plot of a non-aggregated metric map."""
 
-    def __init__(self, metric_map, plot_on_click, **kwargs):
-        self.coords = metric_map.map_data.index.to_frame().values
+    @MetricMapPlot.current_metric_map.setter
+    def current_metric_map(self, value):
+        MetricMapPlot.current_metric_map.__set__(self, value)
+        self.coords = self.current_metric_map.map_data.index.to_frame().values
```

Comment:
Redundant cast `pd.Index` -> `pd.DataFrame`
```suggestion
        self.coords = self.current_metric_map.metric_data[self.current_metric_map.coords_cols].values
```

Reply:
`metric_data` may contain duplicate coordinates, so getting them from `map_data` directly is more clear, IMO

`to_frame` conversion is rather fast, so I would rather leave it as it is